src/library/itunes/itunesdao.cpp

```diff
@@ -84,30 +118,34 @@ bool ITunesDAO::importPlaylistRelation(int parentId, int childId) {
 }
 
 bool ITunesDAO::importPlaylistTrack(int playlistId, int trackId, int position) {
-    QSqlQuery& query = m_insertPlaylistTrackQuery;
+    if (m_isDatabaseInitialized) {
+        QSqlQuery& query = m_insertPlaylistTrackQuery;
 
-    query.bindValue(":playlist_id", playlistId);
-    query.bindValue(":track_id", trackId);
-    query.bindValue(":position", position);
+        query.bindValue(":playlist_id", playlistId);
+        query.bindValue(":track_id", trackId);
+        query.bindValue(":position", position);
 
-    if (!query.exec()) {
-        LOG_FAILED_QUERY(query);
-        return false;
+        if (!query.exec()) {
+            LOG_FAILED_QUERY(query);
+            return false;
+        }
     }
 
     return true;
 }
 
 bool ITunesDAO::applyPathMapping(const ITunesPathMapping& pathMapping) {
-    QSqlQuery& query = m_insertPlaylistTrackQuery;
+    if (m_isDatabaseInitialized) {
```

Comment:
See the comment above regarding the testability, otherwise I'd agree (and this is probably mostly an artifact from `DAO` being designed the way it is)